app/main.py

```diff
@@ -1,52 +1,53 @@
-        self.title = title
-        self.content = content
-
-    def display(self, display_type: str) -> None:
-        if display_type == "console":
-            print(self.content)
-        elif display_type == "reverse":
-            print(self.content[::-1])
-        else:
-            raise ValueError(f"Unknown display type: {display_type}")
-
-    def print_book(self, print_type: str) -> None:
-        if print_type == "console":
-            print(f"Printing the book: {self.title}...")
-            print(self.content)
-        elif print_type == "reverse":
-            print(f"Printing the book in reverse: {self.title}...")
-            print(self.content[::-1])
+def main(book: Book, commands: list[tuple[str, str]]) -> None | str:
+    display_strategies = {
+        "console": ConsoleDisplay(),
+        "reverse": ReverseDisplay()
+    }
+
+    print_strategies = {
+        "console": ConsolePrint(),
+        "reverse": ReversePrint()
+    }
+
+    serialize_strategies = {
+        "json": JSONSerialize(),
+        "xml": XMLSerialize()
+    }
+
+    def display_command(book: Book, method_type: str) -> None:
+        strategy = display_strategies.get(method_type)
+        if strategy:
+            strategy.display(book.content)
         else:
-            raise ValueError(f"Unknown print type: {print_type}")
-
-    def serialize(self, serialize_type: str) -> str:
-        if serialize_type == "json":
-            return json.dumps({"title": self.title, "content": self.content})
-        elif serialize_type == "xml":
-            root = ET.Element("book")
-            title = ET.SubElement(root, "title")
-            title.text = self.title
-            content = ET.SubElement(root, "content")
-            content.text = self.content
-            return ET.tostring(root, encoding="unicode")
+            raise ValueError(f"Unknown display type: {method_type}")
+
+    def print_command(book: Book, method_type: str) -> None:
+        strategy = print_strategies.get(method_type)
+        if strategy:
+            strategy.print_book(book.title, book.content)
         else:
-            raise ValueError(f"Unknown serialize type: {serialize_type}")
+            raise ValueError(f"Unknown print type: {method_type}")
 
+    def serialize_command(book: Book, method_type: str) -> str:
+        strategy = serialize_strategies.get(method_type)
+        if strategy:
+            return strategy.serialize(book.title, book.content)
+        else:
+            raise ValueError(f"Unknown serialize type: {method_type}")
 
-def main(book: Book, commands: list[tuple[str, str]]) -> None | str:
     for cmd, method_type in commands:
         if cmd == "display":
-            book.display(method_type)
+            display_command(book, method_type)
         elif cmd == "print":
-            book.print_book(method_type)
+            print_command(book, method_type)
         elif cmd == "serialize":
-            return book.serialize(method_type)
```

Comment:
No need to create nested functions here. You can even create another class (CommandHandler, for example) to store functions as methods and variables as attributes
But anyway, don't create nested functions in main :)